packages/toolpad-app/src/toolpad/AppEditor/PagesExplorer/index.tsx

```diff
@@ -166,22 +193,73 @@ export default function PagesExplorer({ className }: PagesExplorerProps) {
     if (appDom.isPage(node)) {
       appStateApi.setView({ kind: 'page', nodeId: node.id });
     }
+  };
 
-    if (appDom.isCodeComponent(node)) {
-      appStateApi.setView({ kind: 'codeComponent', nodeId: node.id });
-    }
+  const handlerTreeRef = React.useRef<HTMLUListElement>(null);
+
+  React.useEffect(() => {
+    handlerTreeRef.current?.querySelector(`.${treeItemClasses.selected}`)?.scrollIntoView();
+  }, []);
+
+  const [newPageInput, setNewPageInput] = React.useState('');
+
+  const {
+    value: isCreateNewPageOpen,
+    setTrue: handleOpenCreateNewPage,
+    setFalse: handleCloseCreateNewPageDialog,
+  } = useBoolean(false);
+
+  const nextProposedName = React.useMemo(
+    () => appDom.proposeName(DEFAULT_NEW_PAGE_NAME, existingNames),
+    [existingNames],
+  );
 
-    if (appDom.isConnection(node)) {
-      appStateApi.setView({ kind: 'connection', nodeId: node.id });
+  const handleCreateNewPage = React.useCallback(() => {
+    setNewPageInput(nextProposedName);
+    handleOpenCreateNewPage();
+  }, [handleOpenCreateNewPage, nextProposedName]);
+
+  const handleCloseCreateNewPage = React.useCallback(() => {
+    setNewPageInput('');
+    handleCloseCreateNewPageDialog();
+  }, [handleCloseCreateNewPageDialog]);
+
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(null);
+  const createNewInputRef = React.useRef(null);
+  const open = !!anchorEl;
```

Comment:
`anchorEl` is being reset in an affect when the `newPageInputErrorMsg` turns `null`. This means there is 1 render where `newPageInputErrorMsg` is `null` and `open` is `true`. Could that be the cause of the glitch we see in the video where there is just the icon in the popover?

Reply:
Yeah this code was copied from the functions explorer so I saw that bug but hadn't really fixed it, should be good now.